include/gc/Dialect/Arith/Utils/EasyBuild.h

```diff
@@ -0,0 +1,443 @@
+  }
+
+  template <typename OP, typename V, typename T2, typename Pred>
+  static EBUnsigned handleCmpConst(const T2 &a, const V &b, Pred predicate) {
+    return handleCmp<OP>(EBArithValue::wrap(b.builder, a), b, predicate);
+  }
+
+  template <typename T, typename TOp, typename... Args>
+  static T create(const impl::StatePtr &state, Args &&...v) {
+    return {state,
+            state->builder.create<TOp>(state->loc, std::forward<Args>(v)...)};
+  }
+};
+
+#define DEF_EASYBUILD_BINARY_OPERATOR_FOR_TYPE(OP, OPCLASS, TYPE)              \
+  inline TYPE operator OP(const TYPE &a, const TYPE &b) {                      \
+    return OperatorHandlers::handleBinary<OPCLASS>(a, b);                      \
+  }                                                                            \
+  template <typename T> inline TYPE operator OP(const TYPE &a, T b) {          \
+    return OperatorHandlers::handleBinaryConst<OPCLASS, TYPE>(a, b);           \
+  }                                                                            \
+  template <typename T> inline TYPE operator OP(T a, const TYPE &b) {          \
+    return OperatorHandlers::handleBinaryConst<OPCLASS, TYPE>(a, b);           \
+  }
+
+#define DEF_EASYBUILD_BINARY_OPERATOR(OP, SIGNED, UNSIGNED, FLOAT)             \
+  DEF_EASYBUILD_BINARY_OPERATOR_FOR_TYPE(OP, SIGNED, EBSigned)                 \
+  DEF_EASYBUILD_BINARY_OPERATOR_FOR_TYPE(OP, UNSIGNED, EBUnsigned)             \
+  DEF_EASYBUILD_BINARY_OPERATOR_FOR_TYPE(OP, FLOAT, EBFloatPoint)
+
+DEF_EASYBUILD_BINARY_OPERATOR(+, arith::AddIOp, arith::AddIOp, arith::AddFOp)
+DEF_EASYBUILD_BINARY_OPERATOR(-, arith::SubIOp, arith::SubIOp, arith::SubFOp)
+DEF_EASYBUILD_BINARY_OPERATOR(*, arith::MulIOp, arith::MulIOp, arith::MulFOp)
+DEF_EASYBUILD_BINARY_OPERATOR(/, arith::DivSIOp, arith::DivUIOp, arith::DivFOp)
+DEF_EASYBUILD_BINARY_OPERATOR(%, arith::RemSIOp, arith::RemUIOp, arith::RemFOp)
+
+#undef DEF_EASYBUILD_BINARY_OPERATOR
+#define DEF_EASYBUILD_BINARY_OPERATOR_FOR_INT(OP, SIGNED, UNSIGNED)            \
+  DEF_EASYBUILD_BINARY_OPERATOR_FOR_TYPE(OP, SIGNED, EBSigned)                 \
+  DEF_EASYBUILD_BINARY_OPERATOR_FOR_TYPE(OP, UNSIGNED, EBUnsigned)
+
+DEF_EASYBUILD_BINARY_OPERATOR_FOR_INT(>>, arith::ShRSIOp, arith::ShRUIOp)
+DEF_EASYBUILD_BINARY_OPERATOR_FOR_INT(<<, arith::ShLIOp, arith::ShLIOp)
+DEF_EASYBUILD_BINARY_OPERATOR_FOR_INT(&, arith::AndIOp, arith::AndIOp)
+DEF_EASYBUILD_BINARY_OPERATOR_FOR_INT(|, arith::OrIOp, arith::OrIOp)
+DEF_EASYBUILD_BINARY_OPERATOR_FOR_INT(^, arith::XOrIOp, arith::XOrIOp)
+
+#undef DEF_EASYBUILD_BINARY_OPERATOR_FOR_INT
+#undef DEF_EASYBUILD_BINARY_OPERATOR_FOR_TYPE
+
+inline EBFloatPoint operator-(const EBFloatPoint &a) {
+  return OperatorHandlers::create<EBFloatPoint, arith::NegFOp>(a.builder, a.v);
+}
+
+#define DEF_EASYBUILD_CMP_OPERATOR(OP, OPCLASS, TYPE, PRED)                    \
+  EBUnsigned operator OP(const TYPE &a, const TYPE &b) {                       \
```

Comment:
Missing an `inline` here (also commented on the upstream preparation repo, sorry for spam~)

Reply:
It's not spam! Thanks for finding that! :)